src/buffer.rs

```diff
@@ -71,6 +79,19 @@ impl StreamBuffer {
         twoway::find_bytes(&self.buf, pattern).map(|idx| self.buf.split_to(idx).freeze())
     }
 
+    pub fn read_transport_padding(&mut self) -> bool {
+        match self.buf.iter().enumerate().find(|(_, b)| **b != b' ' && **b != b'\t') {
```

Comment:
Use `self.buf.iter().position()` instead, which does this without the tuple.

I'm always surprised there's no `.find()` on `&[u8]`.